### PR TITLE
add is_deleted function on document

### PR DESCRIPTION
**Pull request by CedricCouton**

### CODE REVIEW

File: couchbase-lite/src/document.rs [hunk 1]
```diff
@@ -130,6 +130,10 @@ impl Document {
     pub(crate) fn exists(&self) -> bool {
         self.inner.as_ref().map(|x| x.exists()).unwrap_or(false)
     }
+
+    pub fn is_deleted(&self) -> bool {
+        self.inner.as_ref().map(|x| x.is_deleted()).unwrap_or(false)
```
Comment (Dushistov) [inline couchbase-lite/src/document.rs:135]: Not sure about such implementation. exists and is_deleted is internal functions,
created with different invariants in mind. They are correct in their usage context,
but would be they are correct if make the public, and change meaning from flag checking
to check that document exists in database, or marked as tombstone in database.

For example in java wrapper they  [code](https://github.com/couchbase/couchbase-lite-java-core/blob/afe4e19f12064ea31b5e63660d7aefe6629acdc8/src/main/java/com/couchbase/lite/Document.java#L186) looks completely different from just flags checking.

Comment (CedricCouton) [reply couchbase-lite/src/document.rs:135]: Ok, so what is your recommandation ? to use the same code as in the java wrapper ?

Comment (Dushistov) [reply couchbase-lite/src/document.rs:135]: What is actual use case? You use `enumerate_all_docs(DocEnumeratorFlags::INCLUDE_DELETED)`,
and need distinguish between tombstone and real documents?

Comment (CedricCouton) [reply couchbase-lite/src/document.rs:135]: It's linked to my other PR #47 , if the internal_get is called with must_exists=false, the deleted doc are returned or not ?

**Pull request closed without merging**